### DISCUSSION TITLE
Installing MountWizzard4 on a Raspberry PI 5 with stellarmate OS

### DISCUSSION THREAD

**Discussion by pdeeskow** [mworion/MountWizzard4#356] [Q&A] [answered]
Dear all,
I am preparing to use a Raspberry PI 5 with StellarMate OS (1.8.2 so far) to control my  setup with a GM2000 Ultraport.

Any hints to install Mountwizard on the PI? Stellarmate OS has python 3.11.2, I have read that install scripts only support python 3.8 or 3.9.  What is recommended: Setup virtual environment with 3.8.x. Or use pyenv?

I am not sure about possible cross effects on KSTARS and INDI.

By the way: I am using Mountwizzard 3 successfully since 2019 on a Windows hardware. But the  old mini PC (2015) seems to have some hardware issues now ....

Thanks for your advice in advance.

Best

Peter

**Comment by mworion** [comment, marked as answer]
Hi Peter,
unfortunately MW4 v3.x only supports python 3.8 - 3.10 due to the dependencies of some underlying libraries. So the simplest solution would be to have 3.10 on your RPI5. 

I'm currently working on a v4.x, which will support up to python 3.12, but so far I did not get precompiled packages for aarch64, which you need for your RPI. I do not know, how long it would take to make a first release.

As I use KStars and INDi as well, I only know about the INDI server, which uses python as well, but in a very loose manner. But I never used a StellarMate device.

Hope this helps.
Michel

**Reply by pdeeskow** [reply]
Thanks for the clarification. Will try to use pyenv to manage parallel python versions on the RaspPi. Best Peter

**Comment by helmbier** [comment]
Hello Michael,
in April you wrote:
“ I'm currently working on a v4.x that supports up to Python 3.12, but so far I haven't gotten any precompiled packages for aarch64 that you need for your RPI."
Is there any new information on this? I'm in the process of getting Linux Mint 22 ready to go in the laptop and Raspi 5 (on the mount), so I'm very interested in Python 3.12 support.

Thanks
Helmar

**Comment by mworion** [comment]
Hi Helmar,
v4.x will support 3.12 and it comes with precompiled packages for arm64 as well. Unfortunately I‘m still not in beta stage. 
Python 3.12 could be sorted out as you could use a python 3.10 in an virtual environment. Still the precompiled packages will be an issue.
Im currently rebuilding the app completely (more changes to do as I originally wanted). As I have hardly any experience with deploying on Raspi5 I would be happy to provide some early versions for you to check if installation and running will work. 
Any thoughts about this ?
Michel

**Comment by helmbier** [comment]
Hello Michael,
I have already thought about pyenv. If the fiddling around can be avoided by using System Python, all the better. First attempts with alternative Python versions have failed. Since I still have a functional MW4 installation with Windows 10, the use of pyenv is not urgent.
My raspi5 is actually fully installed except for MW4: RaspiOS, INDI, KStars, skychart, CCDciel, firecapture and phd2 are running.
If you have alpha versions of MW4 v4.x to test, I would be happy to participate in testing.
Helmar

**Comment by Ruediger3** [comment]
Hello Michel,

any update on MW v4? The problem is, that under Windows Python 3.10.11 is the latest installable version and in any vulnerability scan many security risks are enlisted due to the very old Python version, which has many security flaws. 

Thanks!
Rüdiger

<img width="1326" height="532" alt="image" src="https://github.com/user-attachments/assets/acf0e15a-eacc-49fa-a2a9-affad37292e7" />

<img width="1328" height="346" alt="image" src="https://github.com/user-attachments/assets/aa54eac6-5c28-43ff-a1ed-0d16bc22e2c6" />

**Comment by mworion** [comment]
Hi Rüdiger,

Thanks for asking. I’m still working on it as it will be a technical overhaul in any direction (hopefully without any negative surprises). I guess for the upcoming winter season it will be ready.
Still seeking a solution to support as many python versions as possible on the same time. What did you think about only supporting the latest 3 an forcing the users also to update python every 2 years ?

Michel

**Comment by Ruediger3** [comment]
Hi Michel,
Thanks for the reply. 

I think that is perfectly fine, since older versions do not get any updates - even not security patches. At the moment, it is very tedious to install security patches for python version not supported anymore. E.g. 3.10.11 is the last binary you can install on windows. Fixes have to be compiled on windows by the user himself - this is a huge effort and I doubt anyone is doing that. 

Please go for newest versions only. The old suffer from severe security issues. They compromise the complete windows system.

Many thanks in advance!

Regards
Rüdiger

**Comment by mworion** [comment]
I found a solution not to force everybody on a yearly base to update, so most probably v4 will support the latest 3 python version, so actually 3.11-3.13.

**Comment by Ruediger3** [comment]
That’s really good news! Many thanks for letting us know:
Looking forward to V4😀
